src/rdf4cpp/IRIFactory.cpp

```diff
@@ -226,6 +229,14 @@ nonstd::expected<IRI, IRIFactoryError> IRIFactory::create_and_validate(std::stri
     return IRI::make_unchecked(iri, node_storage);
 }
 
+IRIFactoryError IRIFactory::assign_prefix_checked(std::string_view prefix, std::string_view expanded) {
+    using namespace util::char_matcher_detail;
+    static constexpr auto pattern = PNCharsMatcher | ASCIIPatternMatcher{"."};
```

Comment:
The matcher should follow this: 
![grafik](https://github.com/user-attachments/assets/b75f9eb4-6d18-4374-80c8-f98fdc0c6645)

(https://www.w3.org/TR/sparql11-query/#rPN_PREFIX)

Reply:
The other version was faster. but i guess it does not really matter in this case.